services/src/main/java/org/exoplatform/addons/gamification/listener/social/activity/GamificationActivityListener.java

```diff
@@ -160,23 +160,24 @@ public void saveComment(ActivityLifeCycleEvent event) {
 
     String activityUrl = getActivityUrl(activity);
     Space space = getSpaceOfActivity(parent);
+    boolean isSpaceActivity = space != null;
 
     createActivityGamificationHistoryEntry(activity.getPosterId(),
-                                           activity.getPosterId(),
-                                           GAMIFICATION_SOCIAL_COMMENT_ADD,
+                                           parent.getPosterId(),
+                                           isSpaceActivity ? GAMIFICATION_SOCIAL_COMMENT_ADD_SPACE_STREAM
+                                                           : GAMIFICATION_SOCIAL_COMMENT_ADD_NETWORK_STREAM,
                                            activityUrl);
 
-    boolean isSpaceActivity = space != null;
-    createActivityGamificationHistoryEntry(activity.getPosterId(),
-                                           parent.getPosterId(),
+    createActivityGamificationHistoryEntry(parent.getPosterId(),
+                                           activity.getPosterId(),
                                            isSpaceActivity ? GAMIFICATION_SOCIAL_COMMENT_SPACE_STREAM
                                                            : GAMIFICATION_SOCIAL_COMMENT_NETWORK_STREAM,
                                            activityUrl);
 
     if (space != null) {
       createSpaceGamificationHistoryEntry(space.getPrettyName(),
                                           activity.getPosterId(),
-                                          GAMIFICATION_SOCIAL_COMMENT_ADD,
+                                          GAMIFICATION_SOCIAL_COMMENT_ADD_SPACE_STREAM,
                                           activityUrl);
     }
```

Comment:
This should be removed, no ?

Reply:
No, This gamifies the space its self, gives the space a score , used to detect popular spaces for example